src/drivers/firestore.ts

```diff
@@ -8,35 +8,77 @@ This file is part of @p0security/cli
 
 You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
 **/
-import { config } from "./env";
-import { initializeApp } from "firebase/app";
-import { getAuth } from "firebase/auth";
+import { Identity } from "../types/identity";
+import { tenantConfig } from "./config";
+import { bootstrapConfig } from "./env";
+import { FirebaseOptions, initializeApp } from "firebase/app";
+import {
+  Auth,
+  getAuth,
+  OAuthProvider,
+  SignInMethod,
+  signInWithCredential,
+} from "firebase/auth";
 import {
   collection as fsCollection,
   CollectionReference,
   doc as fsDoc,
   DocumentReference,
   getFirestore,
   terminate,
+  Firestore,
 } from "firebase/firestore";
 
-// Your web app's Firebase configuration
-const firebaseConfig = config.fs;
+let firestore: Firestore;
+let auth: Auth;
```

Comment:
I think we should avoid mutable global variables like these. Esp if they are about authentication. We may make a mistake in updating them and then the user is attempting to log in to the wrong host (= P0 environment).

An idea here is to keep the module-level `const app, firestore, auth` objects and only initialize another firestore when we access the public `orgs` document. That would be a different method, not `getDoc`, but e.g. `getPublicDoc` and it creates its own firestore using the bootstrap config.

Reply:
I was able to move auth to a local variable. But it's a bit tricky to make everything const. Is this a concern even if these are not exported?

Reply:
imo not exported is good enough